lib/AST/Decl.cpp

```diff
@@ -2450,6 +2450,10 @@ DestructorDecl *ClassDecl::getDestructor() {
 }
 
 bool ClassDecl::inheritsSuperclassInitializers(LazyResolver *resolver) {
+  // Get a resolver from the ASTContext if we don't have one already.
+  if (resolver == nullptr)
+    resolver = getASTContext().getLazyResolver();
```

Comment:
It's never going to be incorrect because there can only be one TypeChecker at a time (it asserts if there's more than one). *But* it was a way to catch things that *try to ask AST questions* when they shouldn't, or are assuming the thing has already been type-checked. We've caught bugs that way.